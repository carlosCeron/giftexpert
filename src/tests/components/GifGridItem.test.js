import React from 'react';
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";


describe('Pruebas en GifGridItem', () => {

  let wrapper
  let img

  beforeEach(() => {
    
    img = {
      id: "1",
      title: "Test Component",
      url: "http://url.com/"
    }

    wrapper = shallow(<GifGridItem  {...img} />);

  });

  test('debe demostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe contener un parrafo con el titulo pasado', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(img.title);
  });

  test('debe contener un src y un alt', () => {

    const {src, alt} = wrapper.find('img').props();
    expect(src).toBe(img.url);
    expect(alt).toBe(img.title);
  });

  test('should div has the animated class', () => {
    const div = wrapper.find('div');
    const {className} = div.props();
    expect(className.includes('card')).toBe(true);
  });
  

});