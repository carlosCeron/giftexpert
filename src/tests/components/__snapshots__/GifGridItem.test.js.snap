// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe demostrar el componente correctamente 1`] = `
<div
  className="card card-my-style"
>
  <img
    alt="Test Component"
    src="http://url.com/"
  />
  <p>
    Test Component
  </p>
</div>
`;
